### PR TITLE
Update null safety strictness levels

### PR DESCRIPTION
**Pull request by RealyUniqueName**

### CODE REVIEW

File: content/08-cr-features.md [hunk 1]
```diff
@@ -602,14 +602,36 @@ There are three levels of null safety strictness:
 
 * Off Turn off null safety checks. Useful to selectively disable null safety for particular fields or expression.
 * Loose (Default.) Within an `if (<expr> != null)` condition, `<expr>` is considered safe even if it could be modified after the check.
-* Strict Full-scale null safety checking.
+* Strict Full-scale null safety checking for a single-threaded environment.
+* StrictThreaded Full-scale null safety checking for a multi-threaded environment.
```
Comment (Aurel300) [inline content/08-cr-features.md:606]: The actual cases should be visually separated from the text. I would say backticks and a colon.

 - `Off`: Turn off blah blah
 - `Strict`: Blah blah

File: content/08-cr-features.md [hunk 2]
```diff
@@ -602,14 +602,36 @@ There are three levels of null safety strictness:
...
 
 ```haxe
 @:nullSafety(Off)
 @:nullSafety(Loose)
 @:nullSafety(Strict)
+@:nullSafety(StrictThreaded)
+```
+
+`Strict` and `StrictThreaded` differ in handling of sequential field access.
+In a multi-threaded application sequential access to the same object field may not yeld the same result. That means a null check for a field does not provide any guaranties:
```
Comment (Aurel300) [inline content/08-cr-features.md:618]: "guarantees"

File: content/08-cr-features.md [hunk 3]
```diff
@@ -602,14 +602,36 @@ There are three levels of null safety strictness:
...
+```haxe
+@:nullSafety(StrictThreaded)
+function demo1(o:{field:Null<String>}) {
+  if(o.field != null) {
```
Comment (Aurel300) [inline content/08-cr-features.md:622]:
```suggestion
  if (o.field != null) {
```

File: content/08-cr-features.md [hunk 4]
```diff
@@ -602,14 +602,36 @@ There are three levels of null safety strictness:
...
+    // Error: o.field could have been changed to `null` 
+    // by another thread after the check
+    trace(o.field.length); 
+  }
+}
+
+@:nullSafety(Strict)
+function demo1(o:{field:Null<String>}) {
+  if(o.field != null) {
```
Comment (Aurel300) [inline content/08-cr-features.md:631]:
```suggestion
  if (o.field != null) {
```

File: content/08-cr-features.md [hunk 5]
```diff
@@ -600,16 +600,38 @@ To enable the checker for a particular class, field, or expression, annotate it
 
 There are three levels of null safety strictness:
 
-* Off Turn off null safety checks. Useful to selectively disable null safety for particular fields or expression.
-* Loose (Default.) Within an `if (<expr> != null)` condition, `<expr>` is considered safe even if it could be modified after the check.
-* Strict Full-scale null safety checking.
+* `Off`: Turn off null safety checks. Useful to selectively disable null safety for particular fields or expression.
+* `Loose` (Default): Within an `if (<expr> != null)` condition, `<expr>` is considered safe even if it could be modified after the check.
+* `Strict`: Full-scale null safety checking for a single-threaded environment.
+* `StrictThreaded`: Full-scale null safety checking for a multi-threaded environment.
```
Comment (kaikoga) [inline content/08-cr-features.md:606]: Now we have four levels instead of three?

Comment (RealyUniqueName) [reply content/08-cr-features.md:606]: Thanks. Updated

**Pull request merged**